src/compas_robots/model/robot.py

```diff
@@ -990,6 +1004,181 @@ def _check_link_name(self, name):
         if name in all_link_names:
             raise ValueError("Link name '%s' already used in chain." % name)
 
+    # --------------------------------------------------------------------------
+    # Methods for accessing the visual and collision geometry
+    # --------------------------------------------------------------------------
+
+    def _extract_link_meshes(self, link_elements, meshes_at_link_origin=True):
+        # type: (List[Visual] | List[Collision], Optional[bool]) -> List[Mesh]
+        """Extracts the list of compas meshes from a link's Visual or Collision elements.
+
+        Parameters
+        ----------
+        link_elements : list of :class:`~compas_robots.model.Visual` or :class:`~compas_robots.model.Collision`
+            The list of Visual or Collision elements of a link.
+        meshes_at_link_origin : bool, optional
+            Defaults to True, which means that the meshes will be transformed according to
+            `element.origin`, such that the mesh origin matches with the link's origin frame.
+            If False, the meshes will be extracted as it is loaded from the
+            robot model package. Note that the `.origin` for each element is not necessarily
+            the same.
+
+        Returns
+        -------
+        list of :class:`~compas.datastructures.Mesh`
+            A list of meshes belonging to the link elements.
+
+        Notes
+        -----
+        Only MeshDescriptor in `element.geometry.shape` is supported. Other shapes are ignored.
+
+        """
+        if not link_elements:
+            return None
+
+        meshes = []
+        # Note: Each Link can have multiple visual nodes
+        for element in link_elements:
+            # Some elements may have a non-identity origin frame
+            origin = element.origin.to_compas_frame() if element.origin else Frame.worldXY()
+            t_origin = Transformation.from_frame(origin)
+            # If `meshes_at_link_origin` is False, we use an identity transformation
+            t_origin = t_origin if meshes_at_link_origin else Transformation()
+
+            shape = element.geometry.shape
+            # Note: the MeshDescriptor.meshes object supports a list of compas meshes.
+            if isinstance(shape, MeshDescriptor):
+                # There can be multiple mesh in a single MeshDescriptor
+                for mesh in shape.meshes:
```

Comment:
there is a `LinkGeometry._get_item_meshes` that can be used to handle this, including any primitive shapes.
see example usage in https://github.com/compas-dev/compas_robots/blob/3284fbc4785f99c7871959b14aa9ec67f923ba9b/src/compas_robots/scene/baserobotmodelobject.py#L238

Reply:
Yes, I am aware but I don't like that code path with the coerce to tuple thing.
And I think my code is more extendable if in the future other Descriptors are supported.

Reply:
the code currently doesn't support primitives (i.e. proxies of Sphere, Box etc).

Reply:
To me it would make sense to cover all kinds of link geometries already in this PR, or could you explain why you focus on `MeshDescriptor` and see all others as future extension?

Reply:
Oh, so you mean by using `meshes = LinkGeometry._get_item_meshes(item)` we can cover the other Descriptors too? 

Okay, we can do that too.

Reply:
Fixed.